src/icalendar/cal/todo.py

```diff
@@ -215,11 +215,143 @@ def end(self, end: date | datetime | None):
     def duration(self) -> timedelta:
         """The duration of the VTODO.
 
-        This duration is calculated from the start and end of the Todo.
-        You cannot set the duration as it is unclear what happens to start and end.
+        Returns the DURATION property if set, otherwise calculated from start and end.
+        You can set the duration to automatically adjust the end time while keeping start locked.
         """
+        # First check if DURATION property is explicitly set
+        if "DURATION" in self:
+            # For incomplete todos with only DURATION, we need to be able to compute end
+            # which requires either DTSTART or DUE
+            if self.DTSTART is None and self.DUE is None:
+                # This matches the behavior expected by the test
+                raise IncompleteComponent("No DUE or DURATION+DTSTART given.")
+            return self["DURATION"].dt
+
+        # Fall back to calculated duration from start and end
         return self.end - self.start
 
+    @duration.setter
+    def duration(self, value: timedelta):
+        """Set the duration of the todo.
```

Comment:
This docstring must go into the first declaration. It is not visible in the PR docs but good to have!

https://icalendar--881.org.readthedocs.build/en/881/api.html

Only the getter's doc string will be used for the property doc string.